tests/Validation/ValidationTest.php

```diff
@@ -0,0 +1,196 @@
+            'definitions' => json_decode(json_encode($actualDefs))
+        );
+
+        // Turn references into relative paths
+        foreach ($refsAndDefs['references'] as $key => $list) {
+            $fixedPathRefs = array_map(function ($ref) {
+                global $frameworksDir;
+                return str_replace($frameworksDir, '.', $ref);
+            }, $list);
+
+            $refsAndDefs['references']->$key = $fixedPathRefs;
+        }
+
+        // Turn def locations into relative paths
+        foreach ($refsAndDefs['definitions'] as $key => $def) {
+            if ($def !== null && $def->symbolInformation !== null &&
+                $def->symbolInformation->location !== null && $def->symbolInformation->location->uri !== null) {
+                $def->symbolInformation->location->uri = str_replace($frameworksDir, '.', $def->symbolInformation->location->uri);
+            }
+        }
+
+        return $refsAndDefs;
+    }
+
+    /**
+     * @param $definitions Definition[]
+     * @return array|\array[]
+     */
+    private function getTestValuesFromDefs($definitions): array
+    {
+        // TODO - use reflection to read these properties
+        $propertyNames = ['extends', 'isGlobal', 'isStatic', 'canBeInstantiated', 'symbolInformation', 'type', 'documentation'];
+
+        $defsForAssert = [];
+        foreach ($definitions as $definition) {
+            $fqn = $definition->fqn;
+
+            foreach ($propertyNames as $propertyName) {
+                if ($propertyName === 'symbolInformation') {
+                    // Range is very often different - don't check it, for now
+                    unset($definition->$propertyName->location->range);
+                } elseif ($propertyName === 'extends') {
+                    $definition->$propertyName = $definition->$propertyName ?? [];
+                } elseif ($propertyName === 'type' && $definition->type !== null) {
+                    // Class info is not captured by json_encode. It's important for 'type'.
+                    $defsForAssert[$fqn]['type__class'] = get_class($definition->type);
+                }
+
+                $defsForAssert[$fqn][$propertyName] = $definition->$propertyName;
+            }
+        }
+
+        return $defsForAssert;
+    }
+
+    /**
+     * @return void
+     */
+    private function filterSkippedReferences(&$references)
+    {
+        $skipped = [
+            'false', 'true', 'null', 'FALSE', 'TRUE', 'NULL',
+            '__', // magic constants are treated as normal constants
+            'Exception', 'Error', // catch exception types missing from old definition resolver
+            'Trait', // use Trait references are missing from old definition resolve
+            '->tableAlias', '->realField', '->field', '->first_name', '->last_name', '->quoteMatch', '->idCol', '->timeCol', '->dataCol',
+            'pathToUri', 'uriToPath' // group function use declarations are broken in old definition resolver
```

Comment:
I think this was just to smooth over known differences between the two DefinitionResolvers. It shouldn't be needed for the tests now, removed.